### ISSUE TITLE
Folder resource: Provider produced inconsistent result after apply

### ISSUE THREAD

**Issue by miguelglopes** [grafana/terraform-provider-grafana#665]
### Terraform Version
* Terraform: 1.3.0
* Terraform Grafana Provider: 1.24.0 and 1.28.2
* Grafana: Grafana Cloud

### Affected Resource(s)
* grafana_folder

### Terraform Configuration Files
```hcl
terraform {
  required_providers {
    grafana = {
      source  = "grafana/grafana"
      version = "1.28.2"
    }
  }
}
provider "grafana" {
  # this is for grafana apis. Required for adding dashboards
  org_id = var.GRAFANA_ORG
  url    = var.GRAFANA_URL
  auth   = var.GRAFANA_AUTH

  # this is for synthetic monitoring. required for adding synthetic monitor
  sm_url          = var.GRAFANA_SM_URL
  sm_access_token = var.GRAFANA_SM_TOKEN
}
resource "grafana_folder" "folder_test" {
  title = "test"
}

resource "grafana_dashboard" "dashboard_test" {
  config_json = file("${path.module}/test.json")
  overwrite   = true
  folder      = grafana_folder.folder_test.id
}
```

### Debug Output
![image](https://user-images.githubusercontent.com/23117662/191815608-24873c7d-42d0-4e66-be26-84514b22e318.png)

### Panic Output
![image](https://user-images.githubusercontent.com/23117662/191815592-a69db0be-0980-4dc1-9c6c-1494db932630.png)

### Expected Behavior
Should have created the folder and the dashboard inside the folder.

### Actual Behavior
Folder is created but terraform fails at that step, and doesn't create the dashboard.
![image](https://user-images.githubusercontent.com/23117662/191815466-a0519024-cb0c-4b2f-8aed-0fd93a22cacf.png)

### Steps to Reproduce
1. `terraform apply`

**Comment by aegdevops** [comment]
Same issue
### Versions
**Terraform:** v.1.3.4
**Grafana Provider:** v1.24.0 and v.1.30.0
**Grafana OSS:** v9.2.2

**Comment by julienduchesne** [comment]
Related to https://github.com/grafana/terraform-provider-grafana/issues/521. I will ask around about this. Seems like an Editor is allowed to create a folder but not manage it or, in 9.2.x, even read it. To fix this, use an Admin token. Though this is very much a workaround, a user that creates a folder should probably be allowed to manage it IMO

**Comment by IevaVasiljeva** [comment]
Hi! I've created a fix that I think should resolve this problem: https://github.com/grafana/grafana/pull/59101

It will be released with Grafana 9.3.0 and Grafana 9.2.7. 

Let me know if it doesn't fix the issue for you and I'll take another look at this.

**Some background on the problem:**

My theory is that this issue is caused by permission caching on Grafana's side. When a new resource (ie, a folder or a team) is created, Grafana automatically sets up a bunch of permissions to access this resource. However, permissions are cached, so if a request to read the new resource is sent right after resource creation, the cached permissions will be used and the new permissions will be ignored.

The fix clears permission cache for resource creator, so their updated permissions will be used for their next request.
